powermon/commands/response_definition.py

```diff
@@ -74,63 +74,63 @@ def from_config(cls, response_definition_config : list) -> "ResponseDefinition":
                                             description=response_definition_description,
                                             unit=unit,
                                             extra_info=response_definition_extra)
         
-        elif response_definition_type == ResponseDefinitionType.FLOAT:
+        elif response_definition_type == ResponseType.FLOAT:
             unit = response_definition_config[3]
             return ResponseDefinitionFloat(index=response_definition_index,
                                             description=response_definition_description,
                                             unit=unit,
                                             extra_info=response_definition_extra)
             
-        elif response_definition_type == ResponseDefinitionType.STR_KEYED:
+        elif response_definition_type == ResponseType.STR_KEYED:
             options : dict[str, str] = response_definition_config[3]
             return ResponseDefinitionStrKeyed(index=response_definition_index,
                                             description=response_definition_description,
                                             options=options,
                                             extra_info=response_definition_extra)
             
-        elif response_definition_type == ResponseDefinitionType.ENFLAGS:
+        elif response_definition_type == ResponseType.ENFLAGS:
             flags : dict[str, dict[str, str]] = response_definition_config[3]
             return ResponseDefinitionENFlags(index=response_definition_index,
                                             description=response_definition_description,
                                             enflags=flags,
                                             extra_info=response_definition_extra)
             
-        elif response_definition_type == ResponseDefinitionType.STRING:
+        elif response_definition_type == ResponseType.STRING:
             unit = response_definition_config[3]
             return ResponseDefinitionString(index=response_definition_index,
                                             description=response_definition_description,
                                             unit=unit,
                                             extra_info=response_definition_extra)
             
-        elif response_definition_type == ResponseDefinitionType.FLAGS:
+        elif response_definition_type == ResponseType.FLAGS:
             flags : list[str] = response_definition_config[3]
             return ResponseDefinitionFlags(index=response_definition_index,
                                             description=response_definition_description,
                                             flags=flags,
                                             extra_info=response_definition_extra)
             
-        elif ResponseDefinitionType.INFO in response_definition_type:
+        elif ResponseType.INFO in response_definition_type:
```

Comment:
how do we handle the template type responsetypes? ie where we have an int that needs to be *10 or /1000 or other re-calculations?

Reply:
ResponseDefinitionInfo.translate_raw_response() uses the template. The template is pulled out on line 155 of response_definition.py

Reply:
templates were not just for info responses, info was mostly 'parse part of the response into a new parameter'
there are also 'look at this other response based on this one' and 'the answer needs math applied' types are well

Reply:
I didn't see any responses like that when looking around. If you have one in mind let me know where and I'll add a type for it.

Reply:
they're not in protocols that have been ported yet, \
example from JK04
```
[
   "lookup:'Voltage Cell{:02d}'.format(m['Highest Cell'][0])",
   0,
   "Highest Cell Voltage",
   "V",
],
```
data has a 'Highest Cell' with a value that is a cell number and this looks up the voltage of that cell

another from pi17m058
```
[1, "Battery maximum charge current", "int:r/10", "A"],
```
result is a 0.1A answer, so we divide by 10 to get A, ie 240  0.1A is 24A


prolly dont need to be added yet, but just wanted you to be aware of stuff that will be needed

Reply:
I would like to get rid of the code as strings and put them into methods for easier maintenance. We can add a whole bunch of response types. I'll work on simplifying how much code needs to go into each response type so adding new responses is easy. We can add INT_DIVIDE_BY_10 as a type in this case.